source/World.cpp

```diff
@@ -174,6 +251,22 @@ namespace GTA
 
 		return gcnew Ped(handle);
 	}
+	array<Entity ^> ^World::GetAllEntities()
+	{
+		List<Entity ^> ^ents = gcnew List<Entity ^>();
+		array<int> ^entities = MemoryAccess::GetEntityHandleList();
+
+		for (int i = 0; i < entities->Length; i++)
+		{
+			if (Native::Function::Call<bool>(Native::Hash::DOES_ENTITY_EXIST, entities[i]))
+			{
+				Entity^ ent = (Entity ^)gcnew Prop(entities[i]);
```

Comment:
something like this would be better
if (Native::Function::Call<bool>(Native::Hash::DOES_ENTITY_EXIST, entities[i]))
			{
				if (Native::Function::Call<bool>(Native::Hash::IS_ENTITY_A_PED, entities[i]))
				{
					ents->Add(gcnew Ped(entities[i]));
				}
				else if (Native::Function::Call<bool>(Native::Hash::IS_ENTITY_A_VEHICLE, entities[i]))
				{
					ents->Add(gcnew Vehicle(entities[i]));
				}
				if (Native::Function::Call<bool>(Native::Hash::IS_ENTITY_AN_OBJECT, entities[i]))
				{
					ents->Add(gcnew Prop(entities[i]));
				}
			}